Description: zparse_dsc_comments can crash with invalid dsc_state
 Although zparse_dsc_comments() does check the types of its operands,
 it wasn't checking the return value from dict_find_string properly.
 It was effectively assuming
 that the dictionary parameter was the *correct* dictionary
 and would contain the key/value pair it needed.
 .
 Here we check to see if the key has not been found
 and throw an error if so.
Origin: upstream, http://git.ghostscript.com/?p=ghostpdl.git;h=7ee525f
Author: Ken Sharp <ken.sharp@artifex.com>
Forwarded: yes
Bug: http://bugs.ghostscript.com/show_bug.cgi?id=699801
Last-Update: 2018-10-18
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/psi/zdscpars.c
+++ b/psi/zdscpars.c
@@ -457,6 +457,9 @@
     code = dict_find_string(opDict, dsc_dict_name, &pvalue);
     if (code < 0)
         return code;
+    if (code == 0)
+        return_error(gs_error_undefined);
+
     dsc_state = r_ptr(pvalue, dsc_data_t);
     /*
      * Pick up the comment string to be parsed.
